result/typetests.py

```diff
@@ -0,0 +1,10 @@
+from typing import cast
+
+from .result import Result
+
+
+res1 = Result.Ok('hello')  # type: Result[str, int]
+if res1.is_ok():
+    len(cast(str, res1.ok()))
+else:
+    cast(int, res1.err()) + 2
```

Comment:
This is inelegant for the user. There should probably be methods like `as_ok()` that return a type `T` instead of `Optional[T]`.

The question is whether those should actually fail at runtime if the type is not `T` (analog to a panic), or if it should just be a pure type thing...

What do you think, @tyehle?

Reply:
The `unwrap` method merged in #9 does what is proposed here. Rust also provides `unwrap_err` and `expect_err` which do the corresponding for the error value instead of the ok value, which would be similarly easy to implement.